src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -181,5 +183,43 @@ internal static MsalTokenResponse CreateFromAndroidBrokerResponse(string jsonRes
 
             return msalTokenResponse;
         }
+
+        public void Log(ICoreLogger logger, LogLevel logLevel)
+        {
+            if (logger.IsLoggingEnabled(logLevel))
+            {
+                StringBuilder withPii = new StringBuilder();
+                StringBuilder withoutPii = new StringBuilder();
+
+                withPii.AppendLine("==MsalTokenResponse==");
+                withoutPii.AppendLine("==MsalTokenResponse==");
+
+                withPii.AppendLine($"Error: {Error} ErrorDescription: {ErrorDescription}");
+                withoutPii.AppendLine($"Error: {Error} ErrorDescription: {ErrorDescription}");
+                withPii.AppendLine($"Scopes: {Scope} ");
+                withoutPii.AppendLine($"Scopes: {Scope} ");
+                withPii.AppendLine($"ExpiresIn: {ExpiresIn} RefreshIn {RefreshIn}");
+                withoutPii.AppendLine($"ExpiresIn: {ExpiresIn} RefreshIn {RefreshIn}");
+
+                withoutPii.AppendLine(
```

Comment:
I don't think we should log any token info or client info, and definitely not RTs, in either pii or non-pii logs. Has the guidance on this changed? 

what problem are we trying to solve? which issue is this linked to?

Reply:
There is a linked issue associated with the PR, it is https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2353

I am no logging any tokens, except for ClientInfo which is not a token. I'm only logging true or false if the token is present or not.

Reply:
got it. my mistake, sorry about the false alarm.

Reply:
No prob, I'm happy that someone is looking at this important aspect in detail. Would not want to be the root cause of some security incident :p